mm/2s2h/Enhancements/Actor/BankerDialogue.cpp

```diff
@@ -0,0 +1,41 @@
+#include <libultraship/bridge.h>
+#include "Enhancements/GameInteractor/GameInteractor.h"
+#include "z64.h"
+#include "global.h"
+
+#include "overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.h"
+
+void RegisterBankerDialogue() {
+
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnActorUpdate>([](Actor* actor) {
+        if (!CVarGetInteger("gEnhancements.Actor.BankerDepositRupees", 0)) {
+            return;
+        }
+        if (actor->id == ACTOR_EN_GINKO_MAN) {
+            if (gPlayState->msgCtx.currentTextId == 1104) {
+                if (CHECK_BTN_ALL(gPlayState->state.input[0].cur.button, BTN_Z)) {
+                    char firstChar = (gSaveContext.save.saveInfo.playerData.rupees / 100) + '0';
+                    char secondChar = ((gSaveContext.save.saveInfo.playerData.rupees / 10) % 10) + '0';
+                    char thirdChar = (gSaveContext.save.saveInfo.playerData.rupees % 10) + '0';
+                    const char rupeeChar[3] = { firstChar, secondChar, thirdChar };
+
+                    for (int i = 22; i <= 24; i++) {
+                        gPlayState->msgCtx.bankRupeesSelected = gPlayState->msgCtx.decodedBuffer.schar[i] =
+                            rupeeChar[i - 22];
+                        Font_LoadCharNES(gPlayState, gPlayState->msgCtx.decodedBuffer.schar[i],
+                                         gPlayState->msgCtx.unk120C4 + (0 + (i - 22) << 7));
+                    }
```

Comment:
Hardcoding this logic for character position `22..24`, is prone to not work well with future other languages. Thankfully there is a variable that tracks the "rupee starting position" (`msgCtx.unk120C0`), and you are already using another variable that helps (`msgCtx.unk120C4`). Re-writing like this would be more appropriate and align with how the original code handles things.

```suggestion
                    for (int i = 0; i <= 2; i++) {
                        gPlayState->msgCtx.decodedBuffer.schar[gPlayState->msgCtx.unk120C0 + i] = rupeeChar[i];
                        Font_LoadCharNES(gPlayState,
                                         gPlayState->msgCtx.decodedBuffer.schar[gPlayState->msgCtx.unk120C0 + i],
                                         gPlayState->msgCtx.unk120C4 + (i << 7));
                    }
```